frontend/src/metabase/components/SendTestEmail.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { t } from "ttag";
+
+import ActionButton from "metabase/components/ActionButton";
+
+export default class SendTestEmail extends Component {
+  static propTypes = {
+    channel: PropTypes.object.isRequired,
+    pulse: PropTypes.object.isRequired,
+    testPulse: PropTypes.func.isRequired,
+  };
+  static defaultProps = {};
+
+  onTestPulseChannel(channel) {
+    return this.props.testPulse({ ...this.props.pulse, channels: [channel] });
+  }
+
+  render() {
+    const { channel } = this.props;
+    return (
+      <ActionButton
+        actionFn={this.onTestPulseChannel.bind(this, channel)}
```

Comment:
You can avoid this binding like this if you make `onTestPulseChannel` an [instance field](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields#Public_instance_fields). 

This prop can be `actionFn={this.onTestPulseChannel}` and the function will look like: 

```js
  onTestPulseChannel = () => 
    this.props.testPulse({ ...this.props.pulse, channels: [this.props.channel] });
```

Reply:
Thanks! I was cargo-culting `PulseEditChannels` and am glad I can make things cleaner